### PR TITLE
ARROW-17965: [C++] ExecBatch support for ChunkedArray values

### PR DESCRIPTION
**Pull request by rtpsw**
See https://issues.apache.org/jira/browse/ARROW-17965

### CODE REVIEW

File: cpp/src/arrow/compute/exec.cc [hunk 1]
```diff
@@ -90,14 +90,29 @@ void PrintTo(const ExecBatch& batch, std::ostream* os) {
...
+        auto array = value.make_array();
+        *os << "Array";
+        ARROW_CHECK_OK(PrettyPrint(*array, options, os));
+      } else {
+        auto array = value.chunked_array();
+        *os << "Chunked Array";
+        ARROW_CHECK_OK(PrettyPrint(*array, options, os));
+      }
+      *os << "\n";
+    } else if (value.is_chunked_array()) {
```
Comment (kou) [inline cpp/src/arrow/compute/exec.cc:106]: Duplicated condition?

File: cpp/src/arrow/compute/exec.cc [hunk 2]
```diff
@@ -90,14 +90,29 @@ void PrintTo(const ExecBatch& batch, std::ostream* os) {
     if (value.is_scalar()) {
       *os << "Scalar[" << value.scalar()->ToString() << "]\n";
       continue;
```
Comment (kou) [inline cpp/src/arrow/compute/exec.cc:92]: Should we remove this?

File: cpp/src/arrow/compute/exec.cc [hunk 3]
```diff
@@ -118,8 +133,15 @@ std::string ExecBatch::ToString() const {
 ExecBatch ExecBatch::Slice(int64_t offset, int64_t length) const {
   ExecBatch out = *this;
   for (auto& value : out.values) {
-    if (value.is_scalar()) continue;
-    value = value.array()->Slice(offset, length);
+    if (value.is_scalar()) {
+      continue;
```
Comment (kou) [inline cpp/src/arrow/compute/exec.cc:137]: Should we replace this with `// Do nothing` or something?

File: cpp/src/arrow/compute/exec.cc (hunk 1 above)
Comment (rtpsw) [reply cpp/src/arrow/compute/exec.cc]: Yes, this is an oversight. Will remove.

File: cpp/src/arrow/compute/exec.cc (hunk 2 above)
Comment (rtpsw) [reply cpp/src/arrow/compute/exec.cc]: You mean remove the `continue` statement? I suppose we could.

File: cpp/src/arrow/compute/exec.cc (hunk 3 above)
Comment (rtpsw) [reply cpp/src/arrow/compute/exec.cc]: You mean remove the `continue` statement? I suppose we could.

**Comment by rtpsw** [comment]
@kou, what remains for this PR to go through?

**Review by lidavidm — APPROVED** [review]
LGTM. (Needs to be rebased, though.)

**Comment by rtpsw** [comment]
@lidavidm, it would be easier for me to resolve-conflict directly. OK with you?

**Comment by lidavidm** [comment]
Oh sure, as long as the conflict is solved

**Pull request merged**